### PR TITLE
Feature/enforced binary guild implementation

### LINKED ISSUE #43: EnforcedBinaryGuild - SWPR Guild
For SWPR and potentially other guilds it would be great for us to have a way to enforce a minimum of two options for a guild, whatever the user adds as well as a smart contract enforced action that does nothing, this should be considered the "No" vote. 

We can do this by adding an additional action to the arrays and number of actions in the `createProposal` function

We should also consider the enforced no vote when changing the state of the proposal. If the no vote has "won" then the proposal status should be failed not passed. 

The default way of voting against a proposal in the base guild is simply to not vote however this would add a little more security and make it clear that people are against a proposal whereas before the action would just be to do nothing.

### PR DESCRIPTION
**Pull request by mprasanjith**
A simple Enforced Binary Guild implementation, based on the specification on #43.

- Closes #43.

### CODE REVIEW

File: contracts/erc20guild/implementations/EnforcedBinaryGuild.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+*/
+contract EnforcedBinaryGuild is ERC20Guild {
+    using SafeMathUpgradeable for uint256;
+    using Arrays for uint256[];
+
+    // @dev Create a proposal with an static call data and extra information, and a "No" action enforced.
+    // @param to The receiver addresses of each call to be executed
+    // @param data The data to be executed on each call to be executed
+    // @param value The ETH value to be sent on each call to be executed
+    // @param totalActions The amount of actions that would be offered to the voters
```
Comment (rossneilson) [inline contracts/erc20guild/implementations/EnforcedBinaryGuild.sol:22]: I would maybe include that the total actions is excluding the against vote in this comment

File: contracts/erc20guild/implementations/EnforcedBinaryGuild.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        for (uint256 i = to.length; i < _to.length; i++) {
+            _to[i] = address(0);
+            _data[i] = "";
+            _value[i] = 0;
+        }
+        totalActions += 1;
+
+        return _createProposal(_to, _data, _value, totalActions, title, contentHash);
```
Comment (rossneilson) [inline contracts/erc20guild/implementations/EnforcedBinaryGuild.sol:59]: Chatted with madusha about this and likely we should call the inherited method in the same way on both methods

File: contracts/erc20guild/implementations/EnforcedBinaryGuild.sol [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            _data[i] = data[i];
+            _value[i] = value[i];
+        }
+
+        for (uint256 i = to.length; i < _to.length; i++) {
+            _to[i] = address(0);
+            _data[i] = "";
+            _value[i] = 0;
+        }
+        totalActions += 1;
```
Comment (AugustoL) [inline contracts/erc20guild/implementations/EnforcedBinaryGuild.sol:58]: use safeMath here please

**Review by AugustoL — CHANGES_REQUESTED** [review]

File: contracts/erc20guild/implementations/EnforcedBinaryGuild.sol (hunk 3 above)
Comment (mprasanjith) [reply contracts/erc20guild/implementations/EnforcedBinaryGuild.sol]: Rookie mistake! Fixed.

**Pull request merged**